src/Concordium/Client/Runner.hs

```diff
@@ -2165,14 +2248,15 @@ getWasmModule :: (MonadIO m)
               -> Text -- ^Hash of the block to query in.
               -> ClientMonad m Wasm.WasmModule
 getWasmModule namedModRef block = do
-  res <- getModuleSource (Text.pack . show $ nmrRef namedModRef) block
-
-  case res of
-    Left err -> logFatal ["I/O error:", err]
-    Right (GRPCResponse _ "") -> logFatal [[i|the module reference #{showNamedModuleRef namedModRef} does not exist in block #{block}|]]
-    Right unparsedWasmMod -> case S.decode $ grpcResponseVal unparsedWasmMod of
-      Left err' -> logFatal [[i|could not decode Wasm Module:|], err']
-      Right wasmMod -> return wasmMod
+  bh <- readOrFail block
+  blockRes <- getBlockInfoV2 (Given bh)
+  dieOnRequestFailed ["I/O error:"] blockRes
+  dieOnStatusNotFound [[i|block #{block} does not exist|]] blockRes
+  res <- getModuleSourceV2 (nmrRef namedModRef) (Given bh)
+  dieOnRequestFailed ["I/O error:"] res
+  dieOnStatusNotFound [[i|the module reference #{showNamedModuleRef namedModRef} does not exist in block #{block}|]] res
+  dieOnConversionError ["could not decode Wasm Module:"] res
```

Comment:
I can change it to case-expressions and get rid of the. If it is the check-and-extract pattern which is ugly, it could perhaps be changed to return `[String] -> GRPCResult a -> Either [String] GRPCResult a`, and then sequenced? Then we can `logFatal` on an error? Or is it because case-expressions are altogether better.